beeldengeluid/dane-workflows
# Slack status monitor improvements

fixes https://github.com/beeldengeluid/dane-workflows/issues/13

To test:
- get an appropriate proc_stats db (Philo can help)
- get an appropriate config-example.yaml (Philo can help)
- in `/dane_workflows/` run `python status_monitor.py`
- check that you get expected results displayed in the Slack Channel that is configured in the config, without errors 
- run `scripts/check-project.sh`

NB: some things I would love a reviewer's perspective on:

1. 20 unit tests are failing, I haven't had time to look into this yet
2. not sure if passing PROC_ENV & EXPORTER config values along with status monitor config if the cleanest approach, see https://github.com/beeldengeluid/dane-workflows/commit/365290f6b2c6cb743d712e0446eaabe430bc5bb4
3. the validation of expected config values could be more rigorous, but I have left that for a later enhancement, see https://github.com/beeldengeluid/dane-workflows/commit/12c7b50a70af40b8ddb1f2b9b8b2eff69ea515dc
